### DISCUSSION TITLE
MPS file export

### DISCUSSION THREAD

**Discussion by schmaximilian** [casadi/casadi#3673] [Q&A]
Hi there, 

I am currently using Casadi with HiGHS to solve LP and MILP problems and would like to export my Casadi model using the mps file format. Is there a way to do this or can you give me a workaround? 

Thanks in advance!

**Comment by jaeandersson** [comment]
If your model is an LP or MILP (or QP/MIQP) for that matter, you should be able to get the model out in a standard form:

$$
\begin{array}{cc}
\text{minimize} & \frac{1}{2} x^T \texttt{H} x + \texttt{g}^T x \\
\text{subject to} & \texttt{lbx} \le x \le \texttt{ubx} \\
& \texttt{lba} \le \texttt{A} x \le \texttt{ubA} 
\end{array}
$$

From that data, including sparsity structure for A and H, you should be able to produce such an MPS file.

If you wanted to do it for a nonlinear model, it could be implemented similar to the AMPL interface: https://github.com/casadi/casadi/blob/main/casadi/interfaces/ampl/ampl_interface.cpp#L104
That would be a lot more work, and will only work for simple nonlinear models, so easier to use the approach above if you can.

**Reply by polcrub** [reply]
Isn't there a workaround for intercepting the parametrized .mps (or other format) model sent to the specific solver API when calling a CasAdi solver object, in this case a MILP??

**Reply by jgillis** [reply]
To get the standard form, pass option 'dump_in' to conic/qpsol constructor.
Read in the resulting files with something like `scipy.io.mmread`. From there I'm sure you'll find some python package that can generate MPS format..

**Reply by polcrub** [reply]
Thanks for the rapid response! I'll try it on MATLAB though. This is very useful for tuning solver-based parameters of a model.

**Reply by jgillis** [reply]
In that case, `sparse(casadi.DM.from_file(...))` may be handy to get the standard form dumped with `dump_in`.
